gengokit/gengokit.go

```diff
@@ -47,28 +43,63 @@ type templateExecutor struct {
 	ClientArgs *clientarggen.ClientServiceArgs
 	// A helper struct for generating http transport functionality.
 	HTTPHelper *httptransport.Helper
+	funcMap    template.FuncMap
 }
 
-// GenerateGokit accepts a deftree representing the ast of a group of .proto
-// files, a []truss.NamedReadWriter representing files generated previously, and
-// a goImportPath for templating go code imports
-// GenerateGoCode returns the a []truss.NamedReadWriter representing a generated
-// gokit microservice file structure
-func GenerateGokit(dt deftree.Deftree, previousFiles []truss.NamedReadWriter, goImportPath string) ([]truss.NamedReadWriter, error) {
+// newTemplateExecutor accepts a deftree and a goImportPath to construct a
+// templateExecutor for templating a service
+func newTemplateExecutor(dt deftree.Deftree, goImportPath string) (*templateExecutor, error) {
 	service, err := getProtoService(dt)
 	if err != nil {
-		return nil, errors.Wrap(err, "no service found aborting generating gokit microservice")
+		return nil, errors.Wrap(err, "no service found aborting generating gokit service")
 	}
 
 	importPath := goImportPath + "/" + dt.GetName() + "-service"
-	g := newGenerator(service, importPath, dt.GetName())
-	files, err := g.GenerateResponseFiles(previousFiles, dt.GetName())
+	funcMap := template.FuncMap{
+		"ToLower":    strings.ToLower,
+		"Title":      strings.Title,
+		"GoName":     generatego.CamelCase,
+		"TrimPrefix": strings.TrimPrefix,
+	}
+	return &templateExecutor{
+		ImportPath:  importPath,
+		PackageName: dt.GetName(),
+		Service:     service,
+		ClientArgs:  clientarggen.New(service),
+		HTTPHelper:  httptransport.NewHelper(service),
+		funcMap:     funcMap,
+	}, nil
+}
+
+// GenerateGokit accepts a deftree representing the ast of a group of .proto
+// files, a []truss.NamedReadWriter representing files generated previously,
+// and a goImportPath for templating go code imports. GenerateGokit returns the
+// a []truss.NamedReadWriter representing a generated gokit service file
+// structure
+func GenerateGokit(dt deftree.Deftree, previousFiles []truss.NamedReadWriter, goImportPath string) ([]truss.NamedReadWriter, error) {
```

Comment:
The docstring should mention that `previousFiles` can be nil if there are no previous files.